tools/distpartitioning/dataset_utils.py

```diff
@@ -48,60 +48,171 @@ def get_dataset(input_dir, graph_name, rank, world_size, schema_map):
+            constants.GLOBAL_TYPE_EID, constants.ETYPE_ID]:
+        edge_datadict[col] = []
+
     for etype_name, etype_info in edge_data.items():
-        assert etype_info["format"] == "csv"
+        assert etype_info[constants.STR_FORMAT][constants.STR_NAME] == constants.STR_CSV
 
-        edge_info = etype_info["data"]
+        edge_info = etype_info[constants.STR_DATA]
         assert len(edge_info) == world_size
 
-        data_df = csv.read_csv(edge_info[rank][0], read_options=pyarrow.csv.ReadOptions(autogenerate_column_names=True), 
+        #edgetype strings are in canonical format, src_node_type:edge_type:dst_node_type
+        tokens = etype_name.split(":")
+        assert len(tokens) == 3
+
+        src_ntype_name = tokens[0]
+        rel_name = tokens[1]
+        dst_ntype_name = tokens[2]
+
+        data_df = csv.read_csv(edge_info[rank], read_options=pyarrow.csv.ReadOptions(autogenerate_column_names=True), 
                                     parse_options=pyarrow.csv.ParseOptions(delimiter=' '))
-        edge_datadict[constants.GLOBAL_SRC_ID] = data_df['f0'].to_numpy()
-        edge_datadict[constants.GLOBAL_DST_ID] = data_df['f1'].to_numpy()
-        edge_datadict[constants.GLOBAL_TYPE_EID] = data_df['f2'].to_numpy()
-        edge_datadict[constants.ETYPE_ID] = data_df['f3'].to_numpy()
-
-        v = []
-        edge_file_info = etype_info["data"]
-        for idx in range(len(edge_file_info)):
-            v.append((edge_file_info[idx][1], edge_file_info[idx][2]))
-        edge_tids[etype_name] = v
+        #currently these are just type_edge_ids... which will be converted to global ids
+        edge_datadict[constants.GLOBAL_SRC_ID].append(data_df['f0'].to_numpy() + ntype_gnid_offset[src_ntype_name])
+        edge_datadict[constants.GLOBAL_DST_ID].append(data_df['f1'].to_numpy() + ntype_gnid_offset[dst_ntype_name])
+        edge_datadict[constants.GLOBAL_TYPE_EID].append(np.arange(edge_tids[etype_name][rank][0],\
+                edge_tids[etype_name][rank][1] ,dtype=np.int64))
+        edge_datadict[constants.ETYPE_ID].append(etype_name_idmap[etype_name] * \
+                np.ones(shape=(data_df['f0'].to_numpy().shape), dtype=np.int64))
+
+    #stitch together to create the final data on the local machine
+    for col in [constants.GLOBAL_SRC_ID, constants.GLOBAL_DST_ID, constants.GLOBAL_TYPE_EID, constants.ETYPE_ID]:
+        edge_datadict[col] = np.concatenate(edge_datadict[col])
+
+    assert edge_datadict[constants.GLOBAL_SRC_ID].shape == edge_datadict[constants.GLOBAL_DST_ID].shape
+    assert edge_datadict[constants.GLOBAL_DST_ID].shape == edge_datadict[constants.GLOBAL_TYPE_EID].shape
+    assert edge_datadict[constants.GLOBAL_TYPE_EID].shape == edge_datadict[constants.ETYPE_ID].shape
     print('[Rank: ', rank, '] Done reading edge_file: ', len(edge_datadict), edge_datadict[constants.GLOBAL_SRC_ID].shape)
 
     return node_tids, node_features, node_feature_tids, edge_datadict, edge_tids
```

Comment:
it seems to me that `node_tids` and `node_feature_tids` have redundant information. both of them are used to store the id ranges for each node type.

Reply:
Yes. with the new file format these would be redundant. 
These were introduced during coding the previous version of the input file format. In this format, we had the flexibility of providing different no. of nodes and completely different no. of node features. These two dictionaries where used for this purpose. We can revisit this in the next iteration and change them to our convenience. Lets not make this code change now and target it for the post-release.

Reply:
how about write a TODO here?

Reply:
Done